### ISSUE TITLE
Web UI for editing needles allows creating invalid needles

### ISSUE THREAD

**Issue by ssssam** [os-autoinst/openQA#4815]

**Comment by Martchus** [comment]
> Using drag and drop to expand the green area to the edges is tricky, because the drag ends when the cursor crosses the edge of the screenshot.

That's true. If one moves the mouse too fast out of the screenshot than the area doesn't "catch up" and is therefore not extended to the very edge of the screenshot. When moving the mouse slowly that's not a problem but releasing the drag becomes tricky because one needs still take care to keep the mouse inside the screenshot area.

The easiest workaround is to move the area to the edges first and then extend the other sides. Of course that doesn't work if the area should tangent opposing edges.

I could also reproduce the invalid needles but only in Chromium and not in Firefox. In Chromium one can indeed get 769 for `ypos + height` (and 768 should be the maximum). I suppose the code should explicitly ensure that needles are within the boundaries of the screenshot.

> Another nice to have would be a way to edit the JSON file from the web UI directly, so that dragging the green areas isn't the only way to change them.

That's true. Currently it isn't even possible to C&P from the "Review JSON" button so this needs improvement anyways in my opinion.
